ksw2_extd2_sse.c

```diff
@@ -158,13 +158,21 @@ void ksw_extd2_sse(void *km, int qlen, const uint8_t *query, int tlen, const uin
 		st0 = st, en0 = en;
 		st = st / 16 * 16, en = (en + 16) / 16 * 16 - 1;
 		// set boundary conditions
-		if (st > 0) {
-			if (st - 1 >= last_st && st - 1 <= last_en) {
-				x1 = x8[st - 1], x21 = x28[st - 1], v1 = v8[st - 1]; // (r-1,s-1) calculated in the last round
-			} else {
+		if (st0 > 0) {
+			if (st0 - 1 >= last_st && st0 - 1 <= last_en) {
+				if (r == 160)
+                    printf("this is r == 160, last_st = %d, st0 = %d\n",
+                           last_st, st0);
+                x1 = x8[st0 - 1], x21 = x28[st0 - 1],
+                v1 = v8[st0 - 1];  // (r-1,s-1) calculated in the last round
+            } else {
 				x1 = -q - e, x21 = -q2 - e2;
-				v1 = -q - e;
-			}
+                // v1 = -q - e;
```

Comment:
For entries not in the band (but not outside of the tlen x qlen boundary), extd2_see initialize it to -q-e, instead of 
v1 = r == 0? -q - e : r < long_thres? -e : r == long_thres? long_diff : -e2; 
as described in the Minimap2 paper.

Reply:
Not sure if this is intentional, or a bug of the original minimap2 implementation.